### PR TITLE
Add linter rules from serverless. Minor refactor

### PR DESCRIPTION
**Pull request by andresmgot**
This PR includes the changes required to use the same linter rules than serverless/serverless. It also includes the required fixes to comply with the linter and a minor refactor for some common functions and follow the same style than `serverless/serverless` plugins.

It also removes the duplicated file located at `remove/remove/kubelessRemove.js`

### CODE REVIEW

**Comment by ngtuna** [comment]
LGTM except esLint I have no idea what it's used for

File: deploy/kubelessDeploy.js [hunk 1]
```diff
@@ -12,46 +12,50 @@ class KubelessDeploy {
...
 
-    var thirdPartyResources = new Api.ThirdPartyResources({
-      url: process.env['K8SAPISERVER'],
-      ca: fs.readFileSync(process.env['HOME'] + "/.minikube/ca.crt"),
-      cert: fs.readFileSync(process.env['HOME'] + "/.minikube/apiserver.crt"),
-      key: fs.readFileSync(process.env['HOME'] + "/.minikube/apiserver.key"),
-      group: 'k8s.io',
-    });
+    const thirdPartyResources = new Api.ThirdPartyResources(
+      Object.assign(helpers.getMinikubeCredentials(), {
```
Comment (sebgoa) [inline deploy/kubelessDeploy.js]: let's open an issue to track the case where the k8s endpoint is not minikube but on GKE or some manually setup `kubeadm` cluster. We need to make sure we can get the creds properly. We can do this in a separate PR.

File: deploy/kubelessDeploy.js [hunk 2]
```diff
@@ -12,46 +12,50 @@ class KubelessDeploy {
...
+        url: process.env.K8SAPISERVER,
```
Comment (sebgoa) [inline deploy/kubelessDeploy.js]: and fWIW I don't like this variable name. Maybe KUBE_API_URL ? or KUBE_API_ENDPOINT ? we should try to be consistent with the google cloud function, azure functions and openwhisk variable naming...

File: package.json [hunk 3]
```diff
@@ -18,5 +19,17 @@
   "bugs": {
     "url": "https://github.com/bitnami/kubeless-serverless/issues"
   },
-  "homepage": "https://github.com/bitnami/kubeless-serverless#readme"
+  "homepage": "https://github.com/bitnami/kubeless-serverless#readme",
```
Comment (sebgoa) [inline package.json]: github.com/serverless/serverless-kubeless

File: deploy/kubelessDeploy.js (hunk 1 above)
Comment (andresmgot) [inline deploy/kubelessDeploy.js]: Opened at #8

File: deploy/kubelessDeploy.js (hunk 2 above)
Comment (andresmgot) [inline deploy/kubelessDeploy.js]: My +1 to KUBE_API_URL. Other examples:
https://serverless.com/framework/docs/providers/aws/guide/credentials/#quick-setup
https://serverless.com/framework/docs/providers/openwhisk/guide/credentials/#quick-setup

**Comment by andresmgot** [comment]
I am landing this PR for sending another one based on this. Let me know if you find something else and I will open an new one.
